recipes-domu/domu-image-android/domu-image-android.bb

```diff
@@ -1,76 +1,129 @@
-SUMMARY = "Base for Android images"
+SUMMARY = "Base for build of Android"
 
 LICENSE = "MIT"
 
-inherit build_yocto
-inherit xt_quirks
-inherit xt_reconstruct
+require inc/xt_shared_env.inc
 
-FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
-FILESEXTRAPATHS_prepend := "${THISDIR}/../../classes:"
+# Important note.
+# We use REPODIR and SRC_URI with 'directpath=1' to bypass fetcher's
+# modification of download path. This is done for performance reasons.
+#
+# 'Out-of-stock' repo fetcher gets sources into ${DL_DIR}/repo/xxx folder,
+# then tar/gzip them and then ungzip to ${S}.
+# These steps add almost one hour to fetching on PC.
+#
+# With modifications to xt-distro's repo fetcher (adding 'depth' and 'directpath'
+# parameters to SRC_URI) and using REPODIR instead of DL_DIR (to avoid copying),
+# we have do_fetch() completed within 20 minutes.
+#
+# Some details:
+# 'depth=1' reduces amount of data downloaded from server and is most significant
+#           reducer of fetch time
+# 'REPODIR' is used to download sources directly into required place,
+#           and to avoid copying of tens of gigabytes from ${DL_DIR}
+# 'directpath=1' is used to suppress creation of subfolders to ${REPODIR}.
+#           This is required for correct work of do_compile() with downloaded sources.
 
-S = "${WORKDIR}/repo"
+python __anonymous () {
```

Comment:
i think, the required parameters of url must be checked in the function 'anonymous', and if parameters are not defined or have the unexpected values - stop the processing.